src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentRuntimeNodeWriter.cs

```diff
@@ -255,8 +255,15 @@ public override void WriteHtmlContent(CodeRenderingContext context, HtmlContentI
 
             // Text node
             var content = GetHtmlContent(node);
+            var renderApi = ComponentsApi.RenderTreeBuilder.AddContent;
+            if (content != null && string.IsNullOrWhiteSpace(content))
```

Comment:
Why special-case for when the *entire* text is whitespace? Don't we also want similar handling for when the newlines are somewhere among not-completely-whitespace text?

Reply:
Example:

```
<div>
    The time is @DateTime.Now
</div>
```

... where the first content item would be: `\r\n    The time is `.

Reply:
You are right. See my comment above https://github.com/aspnet/AspNetCore-Tooling/pull/374#discussion_r269853053